src/goto-programs/goto_trace.cpp

```diff
@@ -292,12 +291,11 @@ state_location(const goto_trace_stept &state, const namespacet &ns)
   if(!source_location.get_file().empty())
     result += "file " + id2string(source_location.get_file());
 
-  if(!state.function.empty())
+  if(!source_location.get_function().empty())
   {
-    const symbolt &symbol = ns.lookup(state.function);
     if(!result.empty())
       result += ' ';
-    result += "function " + id2string(symbol.display_name());
+    result += "function " + id2string(source_location.get_function());
```

Comment:
The pretty printing of Java signatures is lost: https://travis-ci.org/diffblue/cbmc/jobs/482137387#L5516

Reply:
Yes, but I'm not entirely sure this is something that is expected. Shouldn't the `source_location.get_function()` equally display a pretty name as that's the name people see in the source code? Maybe it depends on whether we view "Java text source" or "Java bytecode" as the "source." But then line numbers and file names might not make much sense when it's bytecode that we consider as the "source" for the purpose of a "source location."